### PR TITLE
🐛 Esports league page ladder and clan loading

### PR DESCRIPTION
**Pull request by AndrewJakubowicz**
# What does this PR fix?

This PR fixes some esports page bugs.

1. Fixes race condition where two sessions appear next to one another
2. Clear vuex ranking sessions between pages in case the user isn't in that clan or leaderboard.
3. Also stops accidental global session load even when you aren't looking at global version of page.
4. Fixes navigation of clans when clicking on leaderboard or dropdown. As well as browser back and forward arrows.
5. Handle direct navigation to incorrect or malformed clan slug
6. Handle creation of clan without accidental chinese clan name.

# How to test

- Start a dev server connected to the proxy (or production) with `npm run dev` and `npm run proxy`.
- Spy on my account: andrew@codecombat.com
- Navigate to codecombat.com/league . It may take a couple refreshes but you may be able to repro the following screenshot
image

- Navigate to https://codecombat.com/league/5febc4e0b0124ed69f57e07e
- Select a clan in the leaderboard.

image

These issues shouldn't happen with these changes checked out via localhost:3000

# Risk

Low risk. Tested across global leaderboard, joined clan, and not joined clan.

# Screenshots

This screenshot shows what happens when your session state remains when inspecting a clan. We share a single variable in the store for your session between global and clan ranking. We need to ensure that we clear this and only load the data when it is required. Here my session appears in a clan because it is leaking from the global leaderboard.

Screen Shot 2021-01-04 at 4 50 12 PM

Clicking on clans now works (last clan doesn't load in time for the gif):

fab37952a355c0d5d7d0bbe4e45319b8

Pressing the back button is correctly handled:

180c0fef1cbbe3111455abcf052f5de2

Typing in a clan slug or id that doesn't exist is now handled by defaulting to global leaderboard:
45f6400c032967d5e5349dbdf689c83b

### CODE REVIEW

File: app/core/store/modules/seasonalLeague.js [hunk 1]
```diff
@@ -70,7 +74,11 @@ export default {
         splitRankings.push(...state.globalRankings.globalTop.slice(0, 10))
         splitRankings.push({ type: 'BLANK_ROW' })
         splitRankings.push(...state.globalRankings.playersAbove)
-        splitRankings.push(state.mySession)
+        // This hack is due to a race condition where the server returns the player
+        // in the 4 above or 4 below. Thus we prevent player seeing duplicate of their result.
```
Comment (AndrewJakubowicz) [inline app/core/store/modules/seasonalLeague.js:78]: To expand on this, the server is constantly re-scoring players and we fetch your session and the 4 above/below separately. Thus there was a chance that your session appears duplicated.

File: app/views/landing-pages/league/PageLeague.vue [hunk 2]
```diff
@@ -20,7 +20,6 @@
 
       created () {
         this.fetchRequiredInitialData({ optionalIdOrSlug: this.$route.params.idOrSlug })
-        this.loadGlobalRequiredData()
```
Comment (AndrewJakubowicz) [inline app/views/landing-pages/league/PageLeague.vue:23]: Accidentally always loading the global leaderboard even when looking at a clan page.

**Review by codeluggage — APPROVED** [review]
Hard to follow all the branching happening here, but LGTM! Did not test locally.

File: app/core/store/modules/seasonalLeague.js (hunk 1 above)
Comment (nwinter) [reply app/core/store/modules/seasonalLeague.js:78]: Maybe at some point we look into fetching 5 below/above and showing 4, so that even if we have to subtract one, our leaderboard still has an extra one to fill in for it? And/or seeing if Redis can query in a way to do this in 1-2 shots instead of 2-3 shots (since the 4-1-4 logic was written for querying from Mongo)?
